### PR TITLE
Average Submatrix>K (Dynamic Programming)

### LINKED ISSUE #5883: Average Submatrix > K (Dynamic Programming)
## 🚀 Feature

[ C++ ]
For a matrix with non-decreasing elements both along rows (left to right) and columns (top to bottom), 
A "successful submatrix for K" is a square submatrix of A, i.e. a submatrix with n row and n column, for any integer n, such that the average of all the integers in this submatrix is ≥K.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yep!

### PR DESCRIPTION
**Pull request by Suvraneel**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yep

### Description

For a matrix with non-decreasing elements both along rows (left to right) and columns (top to bottom), 
A "successful submatrix for K" is a square submatrix of A, i.e. a submatrix with n row and n column, for any integer n, such that the average of all the integers in this submatrix is ≥K.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
fixes #5883 

@HarshCasper @bhav09  @ankitaggarwal23 @siddharth25pandey 
https://replit.com/@Suvraneel/SubmatrixDP

### CODE REVIEW

**Review by ankitaggarwal23 — COMMENTED** [review]
Don't use cp templates and testcase we don't have online judge to test

**Comment by Suvraneel** [comment]
@ankitaggarwal23  Resolved :)

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

Use proper display message how you are reading and writing outputs

Add input-output prompts to make the program more interactive.

Shift the logic to another function and use main for driver code

File: C-Plus-Plus/dp/avg_submatrix_k.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+For a matrix of order n X m, with non-decreasing elements both along rows and columns, 
+A "successful submatrix for K" is a square submatrix of A, i.e. a submatrix with n row and n column, for any integer n, such that the average of all the integers in this submatrix is ≥ K.
+Find the number of such possible successful submatrices for given K.
+
+Algorithm:
+1. Preprocess matrix A such that every element in the auxiliary matrix contains the sum of all the elements 
+preceding it (row-wise & column-wise)... ie, aux[i][j] stores sum of elements from (0,0) to (i,j).
+2. Thereby, sum of the required submatrix can be found dynamically using : 
+aux(bottom_right) - aux(bottom_left) - aux(top_right) + aux(top_left)
+*******************************************************************************/
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/avg_submatrix_k.cpp]: rewrite it like that
```
/*

Problem Statement:
For a matrix of order n X m, with non-decreasing elements both along rows and columns, 
A "successful submatrix for K" is a square submatrix of A, i.e. a submatrix with n row 
and n column, for any integer n, such that the average of all the integers in this submatrix is ≥ K.
Find the number of such possible successful submatrices for given K.

Algorithm:
1. Preprocess matrix A such that every element in the auxiliary matrix contains the sum of all the elements 
preceding it (row-wise & column-wise)... ie, aux[i][j] stores sum of elements from (0,0) to (i,j).

2. Thereby, sum of the required submatrix can be found dynamically using : 
aux(bottom_right) - aux(bottom_left) - aux(top_right) + aux(top_left)

*/
```

File: C-Plus-Plus/dp/avg_submatrix_k.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Number of successful submatrices for (k = 4) is =>  7
+
+Explanation:
+The following are the seven successful submatrices for k=4:
+⌈3   4⌉
+⌊4   5⌋ with average 4; this matrix occurs only once
+⌈4   5⌉
+⌊5   5⌋ with average 4.75; this matrix also occurs only once
+[4] with average 4; we find this matrix twice in A
+[5] with average 5; we find this matrix 3 times in A
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/avg_submatrix_k.cpp]: Remove this part

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Rest LGTM

**Comment by Suvraneel** [comment]
@ankitaggarwal23 
Added I/O prompts, shifted to modular function, shifted complexities, enhanced description, comments & format
Apologies for the previous mix-up... should suffice now...

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**